os/fs/smartfs/smartfs_smart.c

```diff
@@ -2047,7 +2019,7 @@ int smartfs_rename(struct inode *mountpt, const char *oldrelpath, const char *ne
 #ifdef CONFIG_SMARTFS_USE_SECTOR_BUFFER
 		if (oldentry.dsector == newentry.dsector) {
 			/* We will not use any new entry found, we will overwrite the existing entry */
-			ret = smartfs_writeentry(fs, oldentry, newfilename, type, mode, &newentry);
+			ret = smartfs_writeentry(fs, oldentry, type, mode);
```

Comment:
maybe old entry has old path, this will not change anything. How do you think?

Reply:
We are using the old entry only, so path is OK
I should check the name that will be written however.